lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -440,95 +487,258 @@ def mock_run(args, **kwargs):
+                                            }
+                                        }
+                                    }
+                                },
+                                'subdir12':
+                                {
+                                    'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1dc48d0>
+                                },
+                                'subdir11':
+                                {
+                                    'details': <pbench.server.cache_manager.FileInfo object at 0x7fc4f1d98850>
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            """
+
             m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
             m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
-            m.setattr(Path, "glob", mock_glob)
-            m.setattr(os.path, "getmtime", lambda self: 1681814586.1138802)
-            m.setattr(Path, "is_dir", lambda self: True)
-            m.setattr(Path, "is_file", lambda self: True)
             tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
-            tb.unpack()
-            assert tb.cachemap
-            assert tb.cachemap["A"]["dir1"]["details"].location == Path("/mock/A/dir1")
-            assert (
-                tb.cachemap["A"]["dir1"]["dir2"]["details"].timestamp
-                == "2023-04-18 16:13:06"
-            )
-            assert call == ["tar", "find"]
-            assert tb.unpacked == cache / "ABC" / tb.name
+            tar_dir = TestCacheManager.MockTarball.create_tarball("tar_name")
+            tb.cache_map(tar_dir)
+            assert tb.cachemap["tar_name"]["details"].type == "directory"
+
+            # test get_info with a file path
+            file_info = tb.get_info(Path("tar_name/subdir1/f1.txt"))
+            expected_info = {
+                "location": Path("/tmp/tar_name/subdir1/f1.txt"),
+                "name": "f1.txt",
+                "size": "1024",
+                "type": "file",
+            }
+            assert file_info == expected_info
+
+            # test get_info with a directory path
+            dir_info = tb.get_info(Path("tar_name/subdir1"))
+            expected_info = {
+                "directory": ["subdir13", "subdir12", "subdir11"],
+                "file": ["f1.txt"],
+            }
+            assert dir_info == expected_info
+
+            # test get_info with an empty directory path
+            dir_info = tb.get_info(Path("tar_name/subdir1/subdir11"))
+            expected_info = {
+                "location": Path("/tmp/tar_name/subdir1/subdir11"),
+                "name": "subdir11",
+                "type": "directory",
+            }
+            assert dir_info == expected_info
+
+            # test get_info with a symlink path
+
+            # Note: Need to figure out symlink issue
```

Comment:
I'm thinking that "Note" here should be spelled "FIXME".  😀   (That is, this issue should be resolved prior to merging...or, failing that, we need a bigger flag than just a note.)